### DISCUSSION TITLE
QCefDownloadItem start()参数名含义

### DISCUSSION THREAD

**Discussion by L-Super** [CefView/QCefView#255] [Q&A] [answered]
QCefDownloadItem：
```cpp
  void start(const QString& path, bool useDefaultDialog = true) const;
```
这里注释不全，想确认一下，参数的含义分别是下载存放路径和使用默认对话框？
[QCefDownloadItem::start(const QString& path, bool useDefaultDialog)](https://github.com/CefView/QCefView/blob/19fa70382075335fec2cafccba9f5400a5419cf2/include/QCefDownloadItem.h#L50)

**Comment by tishion** [comment]
> QCefDownloadItem：
> 
> ```c++
>   void start(const QString& path, bool useDefaultDialog = true) const;
> ```
> 
> 这里注释不全，想确认一下，参数的含义分别是下载存放路径和使用默认对话框？ [QCefDownloadItem::start(const QString& path, bool useDefaultDialog)](https://github.com/CefView/QCefView/blob/19fa70382075335fec2cafccba9f5400a5419cf2/include/QCefDownloadItem.h#L50)

对的，已经补充。
如果不想自己实现“保存为”对话框，就可以在QCefView::onNewDownloadItem中直接调用
QCefDownloadItem::start(path, true)


如果需要自己实现保存对话框，可以在QCefView::onNewDownloadItem弹出自己实现的对话框，然后根据用户选择来操作QCefDownloadItem

**Reply by L-Super** [reply]
昨晚尝试了一下，如果指定了path，那打开的对话框文件名将是path名，且存放路径不是path路径。是bug?

**Reply by tishion** [reply, marked as answer]
path必须是包含文件名的全路径，并且路径上的所有目录都必须存在。
比如你传入C:\test\dir1\file.log
窗口显示的时候，会自动打开C:\test\dir1\目录，并且文件名为file.log


如果你传入C:\test\dir1
窗口显示的时候，会自动打开C:\test\目录，并且文件名为dir1
